backend-api/src/functions/asyncActiveSession/tokenService/publicKeyGetter.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  decodeProtectedHeader,
+  importJWK,
+  KeyLike,
+  ProtectedHeaderParameters,
+} from "jose";
+import { errorResult, Result, successResult } from "../../utils/result";
+
+export class PublicKeyGetter implements IPublicKeyGetter {
+  async getPublicKey(
+    jwks: IJwks,
+    jwt: string,
+  ): Promise<Result<Uint8Array | KeyLike>> {
+    const getDecodedProtectedHeaderResult =
+      await this.getDecodeProtectedHeader(jwt);
+    if (getDecodedProtectedHeaderResult.isError) {
+      return getDecodedProtectedHeaderResult;
+    }
+
+    const decodedProtectedHeader = getDecodedProtectedHeaderResult.value;
+
+    const keyId = decodedProtectedHeader.kid;
+    if (!keyId) {
+      return errorResult({
+        errorMessage: "kid not present in JWT header",
+        errorCategory: "CLIENT_ERROR",
+      });
+    }
+
+    const jwk = jwks.keys.find((key) => key.kid === keyId);
```

Comment:
```suggestion
    const jwk = jwks.keys.find((key) => key.kid && key.kid === keyId);
```
If a JWK happens to not include `kid` then line 30 will throw an error.

_I don't think we are checking anywhere else if `kid` is in each JWK . If we are then ignore this message._

Reply:
We're not checking if `kid` is in JWK. I think `.find` returns undefined if cannot find a key, which we check for after this line but I could be wrong so I'll update this

Reply:
I was wrong on this one. I didn't test properly so it was throwing an exception for me. Apologies! 🫨